lib/resty/upstream/healthcheck.lua

```diff
@@ -201,8 +199,14 @@ local function peer_ok(ctx, is_backup, id, peer)
     end
 end
 
+-- shortcut error function for check_peer()
+local report_error = function(sock, ctx, is_backup, id, peer, ...) 
```

Comment:
I'm also thinking that `report_error` might not be very descriptive. Maybe `peer_error` is a better name? Not sure if we should just merge this with `peer_fail` though. They look overlapping.